ml2p/core.py

```diff
@@ -371,6 +374,25 @@ def import_string(name):
     return getattr(mod, classname)
 
 
+class ModelDatasetGenerator:
+    """An interface that allows ml2p-docker to generate dataset within SageMaker."""
+
+    def __init__(self, env):
+        self.env = env
+
+    def generate(self):
+        """Generates and stores a dataset to S3.
+
+        This method should:
+
+        * Read data from source (e.g. S3, Redshift, ...).
+        * Process the dataset.
+        * Write the dataset to S3 (using self.env to determine where to write the data
```

Comment:
It might be nice to have a helper method on this class that does the appropriate upload to S3 dance so that individual model writes don't have to. Something like `.upload_to_s3(path)`.

In particular, it would be nice to ensure data doesn't end up in the wrong place in S3 accidentally.

Reply:
Yeah good idea!